env/work/srw_python/uti_math.py

```diff
@@ -626,7 +637,13 @@ def is_positive(num):
             list_of_roots.append(x[i - 1] + (x[i] - x[i - 1]) / (abs(y[i]) + abs(y[i - 1])) * abs(y[i - 1]))
             positive = not positive
     if len(list_of_roots) >= 2:
-        return abs(list_of_roots[-1] - list_of_roots[0])
+        if not return_as_dict:
+            return abs(list_of_roots[-1] - list_of_roots[0])
+        else:
+            return {
+                'fwhm': abs(list_of_roots[-1] - list_of_roots[0]),
+                'x_range': list_of_roots,
+            }
     else:
         raise Exception('Number of roots is less than 2!')
```

Comment:
OK, done.
BTW, do we really need the "fwhm_scipy(x, y)" function that goes after this one?

Reply:
It can be removed, it's not used anywhere. The function was originally created by Timur with the `scipy` dependency.

Reply:
I removed this function.

Reply:
OK.